keps/sig-network/4631-loadbalancerstatus/README.md

```diff
@@ -0,0 +1,1066 @@
+  reprovisioned, but is continuing to handle traffic according to an
+  older state of the Service during the reprovisioning step.
+
+- `Provisioning=False`, `Serving=True`: The load balancer is
+  functioning normally and handling connections according to the
+  last-seen update to the Service.
+
+#### The `LoadBalancerDegraded` Condition
+
+The `LoadBalancerDegraded` condition, when `True`, indicates that the
+cloud provider is aware that it has provisioned the load balancer in a
+way that may not fully implement the semantics of the Service. For
+example, the service requests session affinity, but the load balancer
+does not implement that feature.
+
+(This is not really a great behavior on the cloud provider's part, but
+this is how cloud providers have traditionally behaved, and some users
+may be relying on this in some cases. The `LoadBalancerDegraded`
+condition allows the load balancer to be _explicit_ about the fact
+that it is doing this. However, a cloud provider may also decided to
+not provision a load balancer in these cases (with
+`LoadBalancerServing` set to `False` with `Reason=Unsupported`).)
+
+If the cloud provider is not intentionally providing a degraded load
+balancer, it should leave this condition unset, which clients should
+treat as meaning `Unknown`. In general it is not possible for a cloud
+provider to say definitively that a load balancer is _not_ degraded,
+since the Service object may be using API fields newer than the
+latest version of the API that the cloud provider knows about.
+
+Most of the well-known `Reason` values for this condition consist of a
+`ServiceSpec` field name followed by "`NotSupported`". Thus,
+`Reason=IPFamiliesNotSupported` implies that the load balancer is
+degraded because of something in `Service.Spec.IPFamilies` that could
+not be implemented.
+
+The well-known `Reason` values for the `LoadBalancerDegraded`
+condition are:
+
+  - `Status=True, Reason=IPFamiliesNotSupported`: The service is
+    dual-stack, but the cloud provider has provisioned a single-stack
+    load balancer.
+
+  - `Status=True, Reason=PortsNotSupported`: The service has multiple
+    ports, but the cloud provider has provisioned a load balancer that
+    does not support all of them (possibly because one of the ports
+    uses an unsupported protocol, or because the cloud provider does
+    not support mixed-protocol load balancers).
+
+  - `Status=True, Reason=LoadBalancerIPNotSupported`: The service
+    requested a specific `LoadBalancerIP`, but the cloud provider does
+    not support this functionality or was not able to use the
+    requested IP. (Note that `LoadBalancerIP` is deprecated, and is
+    not supported by most cloud providers.)
+
+  - `Status=True, Reason=SessionAffinityNotSupported`: The service
+    requested session affinity but the cloud provider has provisioned
+    a load balancer that does not support it.
+
+  - `Status=True, Reason=ExternalTrafficPolicyNotSupported`: The
```

Comment:
I'm not sure on this one.  IPMode=Proxy already covers this, no?

Reply:
IPMode=VIP is necessary for implementing ExternalTrafficPolicy=Local but strictly speaking it's not sufficient. (The source IP could be lost for some other reason.)

Also, `IPMode` was intended as cloud-provider-to-service-proxy communication, while the conditions are cloud-provider-to-end-user communication. So even if they're technically redundant, they exist for different reasons?

I agree that it's not clearly necessary though. I could drop it.